RFC/RFC-9-on-chain-validators.md

```diff
@@ -0,0 +1,273 @@
+
+#### Decentraland Name
+
+The [DCLRegistrar](https://etherscan.io/address/0x2a187453064356c898cae034eaed119e1663acb8) contract has a method to get who is the owner of a specific domain: _`getOwnerOf(subdomain: string)`_.
+
+#### Wearables & Emotes
+
+Each Profile entity stores the wearables and emotes equipped by the user. Wearables are stored under `entity.metadata.avatar.wearables` as an array of items' URNs, and Emotes are stored under `entity.metadata.avatar.emotes` as an array of { slot: string, urn: string }. In both cases, the URNs used are the item's urn, not the token's urn. Checking if a user owns a specific item does not scale, since it would require looping within a contract method. Users that own too many NFTs or collections that are too big could cause this to lag or timeout. Another alternative could be making several RPC calls between the catalyst and the Ethereum nodes, which also would not scale well. We should start storing the token's urn within the profiles to make the access checks possible in a single RPC call. The ERC721 standard has a method `ownerOf(tokenId: uint256)` which serves to get the owner of the NFT.
+
+This document will propose two ideas on how the old profiles can be migrated but an new RFC or ADR should be needed describing the final approach.
+
+- **Alternative 1**
+
+  Use the Decentraland Address to signal profile deployments in order to consider them valid without the need to perform further validations. It can be a new field to the profiles schema to avoid re-deploying the profiles by the Decentraland address which can duplicate the storage. If duplicating the profiles is the only alternative, then old ones can be removed.
+
+- **Alternative 2**
+
+  Build a [Merkle tree with every profile hash](https://github.com/decentraland/content-hash-tree) and submit the root to a smart contract. A static file must be added to the repository with all the proofs needed to check if a deployment is part of the tree. The file with the proofs may weight [~100MB for 1M of profiles](https://github.com/decentraland/content-hash-tree#generate-a-tree-with-content-hashes).
+
+For new profiles, a new ADR with a date to make it effective to start receiving profiles with an extended urn with the token id at the end. This implies changes in the Explorer, Kernel, Catalysts, and URN Resolver. New urn resolvers
+
+- `decentraland:{protocol}:collections-v1:{contract(0x[a-fA-F0-9]+)}:{itemName}`
+- `decentraland:{protocol}:collections-v1:{collection-name}:{itemName}`
+- `decentraland:{protocol}:collections-v2:{contract(0x[a-fA-F0-9]+)}:{itemId}`
+- `decentraland:{protocol}:collections-v1:{contract(0x[a-fA-F0-9]+)}:{itemId}:{tokenId}` -> new
+- `decentraland:{protocol}:collections-v1:{collection-name}:{itemName}:{tokenId}` -> new
+- `decentraland:{protocol}:collections-v2:{contract(0x[a-fA-F0-9]+)}:{itemId}:{tokenId}` -> new
```

Comment:
One question here, is the next right?
- `decentraland......{itemId}:{tokenId}` <- urns for items in profiles
- `decentraland......{itemId}` <- urns for the item saved in the content-server?

Reply:
> `decentraland......{itemId}:{tokenId}` <- urns for items in profiles

urns for NFTs in profile

> decentraland......{itemId} <- urns for the item saved in the content-server?
urns for the item, used when a creator deploys an update to the wearable or emote.

Reply:
👍